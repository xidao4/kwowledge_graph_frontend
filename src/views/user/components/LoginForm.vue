<template>
    <div class="card">
        <div class="title">
            登录 SuperCOIN
        </div>
        <a-form :form="form" :label-col="{ span: 5 }" :wrapper-col="{ span: 19 }" @submit="handleSubmit">
            <a-form-item label="账号" style="display: flex; align-items: center">
                <a-input style="border-radius: 20px; height: 40px"
                         v-decorator="['username', { rules: [{ required: true, message: '请输入账号' }] }]"
                         placeholder="请输入账号"
                         :max-length="15"
                />
            </a-form-item>

            <a-form-item label="密码" style="display: flex; align-items: center">
                <a-input type="password"
                         style="border-radius: 20px; height: 40px"
                         v-decorator="['password', { rules: [{ required: true, message: '请输入密码' }] }]"
                         placeholder="请输入密码"
                         :max-length="15"
                />
            </a-form-item>

            <div style="text-align: center; width: 100%">
                <button class="pan-btn pink-btn">
                    登录
                </button>
            </div>
        </a-form>
    </div>
</template>

<script>
    import AFormItem from "ant-design-vue/es/form/FormItem";
    import { mapActions } from 'vuex';
    export default {
        name: "Login",
        components: {AFormItem},
        data(){
            return {
                formLayout: 'horizontal',
                form: this.$form.createForm(this, { name: 'coordinated' }),
            }
        },
        methods: {
            ...mapActions({
                loginAct: 'login'
            }),
            handleSubmit(e) {
                e.preventDefault();
                this.form.validateFields((err, values) => {
                    if (!err) {
                        this.loginAct(values);
                        console.log('Received values of form: ', values);
                    }
                });
            },
        },
    }
</script>

<style scoped lang="scss">
.card{
    padding: 30px;
    min-width: 400px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    border-radius: 10px;
    background-image: none;
}
.card .title {
    font-size: 30px;
    color: rgba(0, 0, 0, .85);
    font-family: Avenir, 'Helvetica Neue', Arial, Helvetica, sans-serif;
    font-weight: 500;
    position: relative;
    top: 2px;
    margin-bottom: 50px;
}
.ant-form label {
    font-size: 25px;
}
.ant-btn{
    font-size: 16px;
}
.margin-bottom-s {
    margin-bottom: 20px;
}
.margin-bottom-m {
    margin-bottom: 35px;
}
.pan-btn {
    font-size: 14px;
    color: #fff;
    padding: 10px 36px;
    border-radius: 20px;
    border: none;
    outline: none;
    transition: 600ms ease all;
    position: relative;
    display: inline-block;

&:hover {
     background: #fff;

&:before,
&:after {
     width: 100%;
     transition: 600ms ease all;
 }
}

&:before,
&:after {
     content: '';
     position: absolute;
     top: 0;
     right: 0;
     height: 2px;
     width: 0;
     transition: 400ms ease all;
 }

&::after {
     right: inherit;
     top: inherit;
     left: 0;
     bottom: 0;
 }
}
@mixin colorBtn($color) {
    background: $color;

    &:hover {
        color: $color;

        &:before,
        &:after {
            background: $color;
        }
    }
}
.pink-btn {
    @include colorBtn(#E65D6E)
}
</style>
