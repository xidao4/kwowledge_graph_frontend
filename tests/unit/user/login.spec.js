import { mount, createLocalVue } from '@vue/test-utils'
import LoginForm from '@/views/user/components/LoginForm.vue'
import Antd from 'ant-design-vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Antd);
localVue.use(Vuex);


describe('Login', () => {
    let actions;
    let store;
    let wrapper;

    beforeEach(() => {
        actions = {
            login: jest.fn()
        };
        store = new Vuex.Store({
            state: {},
            actions
        });
        wrapper = mount(LoginForm, {
            store,
            localVue
        });
    });
    it('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('当表单校验正确，dispatches "loginAct"', () => {
        let inputs = wrapper.findAll('input');
        for(let i = 0; i < inputs.length; i++){
            let input = inputs.at(i);
            input.element.value = 'test';
            input.trigger('input');
        }
        let form = wrapper.find('form');
        form.trigger('submit');
        expect(actions.login).toHaveBeenCalled();
    });

    it('当表单校验失败，not dispatches "loginAct"', () => {
        let form = wrapper.find('form');
        form.trigger('submit');
        expect(actions.login).not.toHaveBeenCalled();
    })
});
