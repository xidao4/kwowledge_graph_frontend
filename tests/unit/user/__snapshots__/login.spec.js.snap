// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<div class="card">
  <div class="title">
    登录 SuperCOIN
  </div>
  <form class="ant-form ant-form-horizontal">
    <div class="ant-row ant-form-item">
      <div class="ant-col ant-col-5 ant-form-item-label"><label for="coordinated_username" title="账号" class="ant-form-item-required">账号</label></div>
      <div class="ant-col ant-col-19 ant-form-item-control-wrapper">
        <div class="ant-form-item-control"><span class="ant-form-item-children"><input placeholder="请输入账号" type="text" maxlength="15" data-__meta="[object Object]" data-__field="[object Object]" id="coordinated_username" class="ant-input" style="border-radius: 20px; height: 40px;"></span>
          <transition-stub appear="true"></transition-stub>
        </div>
      </div>
    </div>
    <div class="ant-row ant-form-item">
      <div class="ant-col ant-col-5 ant-form-item-label"><label for="coordinated_password" title="密码" class="ant-form-item-required">密码</label></div>
      <div class="ant-col ant-col-19 ant-form-item-control-wrapper">
        <div class="ant-form-item-control"><span class="ant-form-item-children"><input placeholder="请输入密码" type="password" maxlength="15" data-__meta="[object Object]" data-__field="[object Object]" id="coordinated_password" class="ant-input" style="border-radius: 20px; height: 40px;"></span>
          <transition-stub appear="true"></transition-stub>
        </div>
      </div>
    </div>
    <div style="text-align: center; width: 100%;"><button class="pan-btn pink-btn">
        登录
      </button></div>
  </form>
</div>
`;
